invoiceregistry_test.go

```diff
@@ -320,6 +320,39 @@ func TestNoSubscriberFail(t *testing.T) {
 	c.checkHtlcFailed(ResultNoAcceptSubscriber)
 }
 
+// TestGracePeriodForSubscriber asserts that a new set is started even if the last subscriber has recently disconnected.
+// After the grace period, we should not accept any new sets
+func TestGracePeriodForSubscriber(t *testing.T) {
+	defer test.Timeout()()
+	t.Parallel()
+
+	// If we reach the accept timeout, it means we've accepted the set_id
+	c := newRegistryTestContext(t, optAcceptTimeout(0))
```

Comment:
I used AcceptTimeout instead of directly checking if our setid is in `htlcSets` because I don't like
Time.Sleep() in tests

Reply:
Not completely sure about the test as is. I understand your wish to not use sleep, but it seems you now rely on workarounds that may not be the easiest to follow.

Perhaps it's time to add a fake clock to mux now. It's in the `lnd/clock` package.